src/frontend/src/btc/services/worker.btc-wallet.services.ts

```diff
@@ -0,0 +1,56 @@
+import { syncWallet } from '$btc/services/btc-listener.services';
+import { isNetworkIdBTCMainnet, isNetworkIdBTCTestnet } from '$icp/utils/ic-send.utils';
+import type { WalletWorker } from '$lib/types/listener';
+import type { PostMessage, PostMessageDataResponseBtcWallet } from '$lib/types/post-message';
+import type { Token } from '$lib/types/token';
+import { mapToSignerBitcoinNetwork } from '$lib/utils/network.utils';
+
+export const initBtcWalletWorker = async ({
+	id: tokenId,
+	network: { id: networkId }
+}: Token): Promise<WalletWorker> => {
+	const WalletWorker = await import('$btc/workers/btc-wallet.worker?worker');
+	const worker: Worker = new WalletWorker.default();
+
+	worker.onmessage = async ({
+		data
+	}: MessageEvent<PostMessage<PostMessageDataResponseBtcWallet>>) => {
+		const { msg } = data;
+
+		switch (msg) {
+			case 'syncBtcWallet':
+				syncWallet({
+					tokenId,
+					data: data.data as PostMessageDataResponseBtcWallet
+				});
+				return;
+		}
+	};
+
+	return {
+		start: () => {
+			worker.postMessage({
+				msg: 'startBtcWalletTimer',
+				data: {
+					bitcoinNetwork: mapToSignerBitcoinNetwork({
+						network: isNetworkIdBTCMainnet(networkId)
```

Comment:
I always feel like it's safer to default to mainnet / production rather than testnet. Always fear that something goes sideways in the build and we endup having test on prod. WDYT?

Reply:
Yes, it does make sense. I adjusted it accordingly.